polkadot/xcm/xcm-builder/src/matcher.rs

```diff
@@ -89,6 +89,17 @@ pub trait MatchXcm {
 		Self: Sized,
 		F: FnMut(&mut Self::Inst) -> Result<(), Self::Error>;
 
+	/// Like `match_next_inst` but it only tries to match on the instruction if `cond`
+	/// is true.
+	///
+	/// If the condition doesn't match, it will not increase the index, which means
+	/// the next function has to deal with the same instruction.
+	fn match_next_inst_if<C, F>(self, c: C, f: F) -> Result<Self, Self::Error>
```

Comment:
no longer needed, but could also be kept as useful for future cases 

haven't reviewed them though

Reply:
Removing them